tonic/models/vic/grid_params.py

```diff
@@ -1648,3 +2150,369 @@ def lake(lake_file, soil_dict, max_numnod=10,
+        CInter = np.zeros((cells, veg_classes_state, snow_bands))
+        CSlow = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_age = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_melt_state = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_coverage = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_water_equivalent = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_surf_temp = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_surf_water = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_pack_temp = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_pack_water = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_density = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_cold_content = np.zeros((cells, veg_classes_state, snow_bands))
+    Snow_canopy = np.zeros((cells, veg_classes_state, snow_bands))
+    Soil_node_temp = np.zeros((cells, veg_classes_state, snow_bands, soil_nodes))
+    if lakes:
+        Lake_soil_moisture = np.zeros((cells, nlayers))
+        Lake_soil_ice = np.zeros((cells, nlayers, nfrost))
+        if carbon:
+            Lake_CLitter = np.zeros((cells))
+            Lake_CInter = np.zeros((cells))
+            Lake_CSlow = np.zeros((cells))
+        Lake_snow_age = np.zeros((cells))
+        Lake_snow_melt_state = np.zeros((cells))
+        Lake_snow_coverage = np.zeros((cells))
+        Lake_snow_water_equivalent = np.zeros((cells))
+        Lake_snow_surf_temp = np.zeros((cells))
+        Lake_snow_surf_water = np.zeros((cells))
+        Lake_snow_pack_temp = np.zeros((cells))
+        Lake_snow_pack_water = np.zeros((cells))
+        Lake_snow_density = np.zeros((cells))
+        Lake_snow_cold_content = np.zeros((cells))
+        Lake_snow_canopy = np.zeros((cells))
+        Lake_soil_node_temp = np.zeros((cells, soil_nodes))
+        Lake_active_layers = np.zeros(cells).astype(int)
+        Lake_layer_dz = np.zeros((cells))
+        Lake_surf_layer_dz = np.zeros((cells))
+        Lake_depth = np.zeros((cells))
+        Lake_layer_surf_area = np.zeros((cells, max_numnod))
+        Lake_surf_area = np.zeros((cells))
+        Lake_volume = np.zeros((cells))
+        Lake_layer_temp = np.zeros((cells, max_numnod))
+        Lake_average_temp = np.zeros((cells))
+        Lake_ice_area_frac = np.zeros((cells))
+        Lake_ice_area_frac_new = np.zeros((cells))
+        Lake_ice_water_equivalent = np.zeros((cells))
+        Lake_ice_height = np.zeros((cells))
+        Lake_ice_temp = np.zeros((cells))
+        Lake_ice_snow_water_equivalent = np.zeros((cells))
+        Lake_ice_snow_surf_temp = np.zeros((cells))
+        Lake_ice_snow_pack_temp = np.zeros((cells))
+        Lake_ice_snow_cold_content = np.zeros((cells))
+        Lake_ice_snow_surf_water = np.zeros((cells))
+        Lake_ice_snow_pack_water = np.zeros((cells))
+        Lake_ice_snow_albedo = np.zeros((cells))
+        Lake_ice_snow_depth = np.zeros((cells))
+
```

Comment:
Good idea - you're right, it *was* rather tedious to code that up, and of course leaves room for errors.  I'll look into it - maybe I can re-use the `dict` elsewhere too.